### DISCUSSION TITLE
Are we sure that copying Webflow's style panel is a good idea?

### DISCUSSION THREAD

**Discussion by mattnaus** [webstudio-is/webstudio-community#7] [Ideas]
First off, let me start by congratulating you on an excellent initiative and starting something that, hopefully sooner rather than later, will be tool that's missing from the current selection of NC tools.

I just want to raise some issues around the intention to move forward with replicating Webflow's styling panel. Perhaps it would be a good idea to apply some first principles thinking here before going down that road? I have worked with a ton of designers over the years and a good amount of them do not like, or feel at home, with the Webflow styling panel. In my opinion, it works great for those with a working understanding of CSS, but that might not necessarily apply to a typical designer who's comfortable with Figma or Sketch. I would at least see if you could get some designers involved in the process (I'd be happy to help getting some involved). 

Something else worth considering is this project's longevity. Has there been any consideration for adding a commercial component to this project, simply to ensure it'll still be around in a couple of years (and actively maintained/developed)? Perhaps something akin to other succesfull OS projects, like Ghost? It would have to be something which does not undermine the projects main goals of course.

I have been building software and software businesses for close to two decades, typically doing most of my development work in the code editor. After selling my last company, I had super high hopes for the nocode industry as I was reluctant spending most of my time in the code editor again. However, after spending a couple of months in the trenches, I realized that most of the existing tools do not meet my requirements (no vendor lock-in, extendable, working with existing codebases, framework agnostic, etc). Webstudio seems to check all those boxes, which makes me super excited! Would love to figure out a way to get involved with this project.

Greetings, 

Matt

**Comment by kof** [comment]
Hi Matt, love the comment!

Replicating Webflow's style panel: the first principle thinking that is guiding me is the philosophy to be a visual development tool, meaning I am manipulating visually the underlying platform, which is CSS. If I go the road of making it more digestible or more simple for various skillsets: e.g. designer persona or user without pre-existing design or coding skills, I will have to create an additional abstraction on top of CSS, which would simplify CSS in one way or another. I was thinking to address this by offering at some point an optional style panel view. Right now the goal is to be able to do anything CSS can do and Webflow is just serving as a good example of how things can be arranged. I am definitely happy to discuss this in-depth and maybe we can come up with something that is great for everyone. It's going to be a process, where I hope to involve as many designers as possible.

Project longevity: I am raising funds right now. There is a planned business model in place: hosted service and additional features for companies.

If you want to chat I added [discord](https://discord.gg/UNdyrDkq5r). Depending on how you want to get involved, we can find different ways. If it's building out features, a good start for everyone will be creating issues, describing them well and prioritizing them.

**Reply by mattnaus** [reply]
Thanks @kof , that makes a ton of sense. I actually had a discussion with a designer friend about this very topic last week; where I was convincing him to learn the basic concepts of HTML and CSS, as it I believed it would make him do web design much more efficiently and it would improve the collaboration between designers and developers. 

I think you're right about this; the fundamentals ARE HTML/CSS and adding additional abstractions on top of those fundamentals to make it more relatable for designers (and non-tech users in general) can easily end up being quite problematic. I guess the ideal approach would be to build such an UI that makes it easy for non-tech users to create and at the same time develop an understanding of underlying fundamentals (HTML&CSS). 

This whole endeavor is indeed going to be a process; fully agree. And it's unlikely going to be "right" on the first iteration.

I will join Discord shortly. Would also love to hear more about the fundraising efforts (depending on the specifics, I might be interested in participating as an investor as well).

**Comment by chris77erickson** [comment]
Matt makes some excellent points but I love the style panel. From talking with friends and colleagues in the design space, it seems that people enjoy designing either in a Figma-esque type environment (like Framer) **or** a style panel setup (like Webflow, WeWeb, YCode). For what it's worth, I had a _very basic_ understanding of HTML/CSS before Webflow and using their style panel really helped me to understand it. Now I feel comfortable building pages without any visual development help. I think the WF approach allows someone to learn how these pages are constructed, much more so than a true drag and drop experience. 

The makers of WebStudio ultimately have to decide what type of customers they want to target. Do you want to go after the Squarespaces/Wix of the world by lowering the (technical) barrier to entry or do they want to go after Wordpress, Webflow, Ycode type designers that are looking for an alternative. I don't know the answer but clearly I'm biased because I'm in the latter group.

**Comment by kof** [comment]
@chris77erickson your are absolutely right. We are not building a tool to build a site in 15 min, we build a tool to turn you into a builder and eventually web developer. Webstudio is where your career should start no matter if you are an experienced designer or just starting with web development

**Comment by kof** [comment]
We have exciting plans about making this tool a great way to learn. Starting with the tooltips and GPT assisted learning.
